public/src/managers/ServerProvider.ts

```diff
@@ -243,6 +243,67 @@ export class ServerProvider implements IUserProvider, ICourseProvider {
         return JSON.parse(JSON.stringify(result.statusCode)) as IStatusCode;
     }
 
+    // getAllGroupLabInfos 
+    public async getAllGroupLabInfos(courseId: number, groupID: number): Promise<IMap<ISubmission>> { 
+        const result = await this.helper.get<ISubmission[]>(
+            ("courses/" + courseId.toString() + "/groups/" + groupID + "/submissions"),
+        );
+        if (result.statusCode === 200 && result.data === undefined) {
+            return {};
+        }
+        if (!result.data) {
+            this.handleError(result, "getAllLabInfos");
+            return {};
+        }
+        return mapify(result.data, (e) => {
+            let a = "{\"builddate\": \"2017-07-28\", \"buildid\": 1, \"buildlog\": \"This is cool\", \"execTime\": 1}";
+            let b = "[{\"name\": \"Test 1\", \"score\": 3, \"points\": 4, \"weight\": 100}]";
+            if ((e as any).buildinfo && ((e as any).buildinfo as string).trim().length > 2) {
```

Comment:
First, I would prefer if all variables were camel cased consistently: `buildInfo` and below `scoreObjects`. PS: Do you mean `scoreObject` here (without the `s`)??

Moreover, this code is not entirely obvious to me; I think it is extracting the `buildinfo` and `scoreobjects` fields from object`e`. I'm sure it is probably doing the right thing, but it looks like this is a pattern (repeated also below), and I wonder if this could be made into a function with a descriptive name, or if the type casting to `string` were done separately. The casting seems to be done three times per object. Alternatively, add a brief comment about what is going on here. But I'm also ok with keeping this code as is, if this is the "appropriate" typescript pattern for doing this.